src/datachain/lib/pytorch.py

```diff
@@ -75,6 +80,17 @@ def __init__(
         if (prefetch := dc_settings.prefetch) is not None:
             self.prefetch = prefetch
 
+        if self.cache or not self.prefetch:
+            self._cache = catalog.cache
+        else:
+            tmp_dir = catalog.cache.tmp_dir
+            assert tmp_dir
+            self._cache = get_temp_cache(tmp_dir, prefix="prefetch-")
```

Comment:
In `PytorchDataset`, we create a temporary cache ourselves and pass it to `rows_prefetcher` instead of letting it do that for us.

This is done because of multiprocessing - each worker would end up creating a separate cache directory.

The downside of this approach is that we'll have to remove the temporary cache ourselves. But we cannot do it at the end of the iteration because `DataLoader` might be keeping a buffer of rows.
So this has to be done manually by users. For this, I have added a `close()` API below.